### DISCUSSION TITLE
Replace the text from pdf saving the structure of the document

### DISCUSSION THREAD

**Discussion by NikitaIVll** [py-pdf/pypdf#2652] [Q&A]
I am attempting to use pypdf to translate PDF files while preserving the formatting and structure of the document. Could you please provide an example of how I can extract text and replace it with the translated text?

**Comment by pubpub-zz** [comment]
replacing text is very tricky : pdf is not a text but e-printed paper. The best that can be done would  be to prepare pages using some other tools (ghostscript, print to pdf, fpdf2,...) and erase to original page content with remove_text() and remove_image() and/or with update_contents(None) then merge the new content in
It is not exactly what you may need but you should inspire from #2648
